src/Workspaces/Core/Portable/Remote/RemoteFeatureOptions.cs

```diff
@@ -25,6 +23,17 @@ public static bool ShouldComputeIndex(Workspace workspace)
                 case WorkspaceKind.RemoteWorkspace:
                     // Always compute indices in the remote workspace.
                     return true;
+
+                case WorkspaceKind.Host:
```

Comment:
so, you are basically doing what my initial PR did that got closed.

if you are checking existence of remote host to do indexing, you dont need switch in ShouldComputIndex per workspace kind.

just always check remote host existence, if it doesn't return true otherwise, return false.

Reply:
you can introduce new IsEnabled or you could just do Workspace.TryGetRemoteHostClientAsync and see whether it returns null.

Reply:
>just always check remote host existence, if it doesn't return true otherwise, return false.

It wasn't clear to me whether temporary and preview workspaces would/should create the index.